### ISSUE TITLE
[TorchToLinalg] Conversion fails to legalize 'torch.constant.int' / No conversion of tm_tensor.attention ops

### ISSUE THREAD

**Issue by DavidGinten** [llvm/torch-mlir#4302]
Hello,

I run a sentence-transformer model through the Torch mlir to Linalg mlir conversion pipeline, but I cannot convert the tm_tensor.attention ops. I tried to decompose the attention ops with PyTorch's decomposition as suggested here #4279 and with that no attention ops appear anymore. However, setting the output type of fx.export_and_import to LINALG_ON_TENSORS yields this error: failed to legalize operation 'torch.constant.int'. More specifically, when first converting to TORCH and then running the following passes:
`torch-mlir-opt
  -canonicalize
  -torch-backend-to-linalg-on-tensors-backend-pipeline \
  model_torch.mlir > model_linalg.mlir`

 I get 

`error: failed to legalize operation 'torch.constant.int'
<unknown>:0: note: see current operation: %120 = "torch.constant.int"() <{value = -1 : i64}> : () -> !torch.int` (I tried different passes here)

How I export the model:
```python
sentences = ["This is an example sentence", "Each sentence is converted"]
tokenizer = AutoTokenizer.from_pretrained('sentence-transformers/all-MiniLM-L6-v2')
model = AutoModel.from_pretrained('sentence-transformers/all-MiniLM-L6-v2')

encoded_input = tokenizer(sentences, padding=True, truncation=True, return_tensors='pt')
class Wrapper(torch.nn.Module):
    def __init__(self, model):
        super().__init__()
        self.model = model
    def forward(self, input_ids, attention_mask):
        outputs = self.model(input_ids, attention_mask=attention_mask)
        return outputs.last_hidden_state

wrapped_model = Wrapper(model)

ep = torch.export.export(
    wrapped_model,
    (
        encoded_input["input_ids"],
        encoded_input["attention_mask"],
    )
)

ep = ep.run_decompositions()
m = fx.export_and_import(ep, output_type=OutputType.LINALG_ON_TENSORS, 
                            func_name = "transformer_model")

mlir_str = str(m)
with open("model_linalg.mlir", "w") as f:
    f.write(mlir_str)
```

I also tried #3461 (Then I don't need the PyTorch decomposition), but it doesn't get rid of the tm_tensor.attention ops.

Thanks for any help/advise in advance.

**Comment by momchil-velikov** [comment]
I've seen this quite a few times and did some (small amount of) investigation. From what I can tell, two transformations undo each other - `ConvertTorchToArith` transforms `torch.constant.int` into `arith.constant` and then canonicalization materializes it back into `torch.constant.int`.

**Comment by DavidGinten** [comment]
Sorry for the delay. @momchil-velikov thanks for your answer, that's true. One can see that when running e.g.
```
echo 'module {
  func.func @torch_constant_int() -> !torch.bool {
    %int_1 = torch.constant.bool true
    return %int_1 : !torch.bool
  }
}' | torch-mlir-opt \
  -pass-pipeline="builtin.module(torch-backend-to-linalg-on-tensors-backend-pipeline)" \
  -mlir-print-ir-after-all -mlir-disable-threading

```
However, the `FuncBackendTypeConversion` pass adds an torch_c.to_i1 op that after canonicalization yields the arith.constant again. So it works then. The problem seems to be that an `torch.aten.any.dim %0, %int-1, %false : ` op is not lowered and therefore we don't get the torch_c.to_i1 op that would result in an arith.constant. Therefore, the `torch-finalizing-backend-type-conversion` pass afterwards fails with the mentioned error. The full pipeline of `-torch-backend-to-linalg-on-tensors-backend-pipeline` is [here](https://github.com/llvm/torch-mlir/blob/ac33bab4f3c1071d12b6c4c2f26bbcf87d1d2ab9/lib/Dialect/TorchConversion/Transforms/Passes.cpp#L69)

**Comment by DavidGinten** [comment]
Solved by [#4467](https://github.com/llvm/torch-mlir/pull/4467)
